packages/flutter_tools/lib/src/context_runner.dart

```diff
@@ -124,7 +124,22 @@ Future<T> runInContext<T>(
         client: globals.httpClientFactory?.call() ?? HttpClient(),
       ),
       DevFSConfig: () => DevFSConfig(),
-      DeviceManager: () => FlutterDeviceManager(),
+      DeviceManager: () => FlutterDeviceManager(
```

Comment:
wow...

Reply:
yeah, I think it might be better to pull out each discovery instance, and then inject them into the constructor? Then the devicemanager is just responsible for ... managing the devices, and not creating the world

Reply:
this will have to wait for a google3 roll, there are some updates needed internally